### DISCUSSION TITLE
Retrieving form submission values

### DISCUSSION THREAD

**Discussion by Camherb** [Budibase/budibase#13815] [Help] [answered]
I've created a form where you enter a latitude, longitude, and radius (in miles). I have connected the application to a Google sheet with a bunch of companies that each have a column for latitude and longitude. When the form is submitted I would like to have it retrieve all of the companies within the specified radius (using the Haversine formula). I added automation with the App Action trigger and connected the form submission button to this automation. I added a JS Scripting block after this with my code to do the haversine formula. I am confused about how I access the values submitted to the form so I can use them in my JS Scripting block.

**Comment by mikesealey** [comment, marked as answer]
Hi @Camherb 

When you create a new automation you should click "Add Field" - in your case they're likely to be `latitude` and `longitude`, as Numbers. 
![image](https://github.com/Budibase/budibase/assets/39269380/64ad8217-bc43-4c2b-9003-02b4da4b4e10)

Now if you're using an Action to trigger the automation, you'll see that you're asked to specify the latitude and longitude with which you're triggering your Automation. In this example, they're captured with a form, so I've set the from the bindings menu.
![image](https://github.com/Budibase/budibase/assets/39269380/5e34a74e-6714-46c1-bc3e-315d9836a3b2)

Now, clicking the button that triggers the automation will pass those two values into the Automation flow. We can then make the next step in the automation a JS Scripting block.  You'll see that the JS Scripting block has bindings available on the right...
![image](https://github.com/Budibase/budibase/assets/39269380/6d66ce73-eaef-4a48-8d8e-958688ae2efe)


From which you can fetch your variables.
![image](https://github.com/Budibase/budibase/assets/39269380/5dddbe99-69a3-4915-bc02-93f2f408657a)

You can also use the results from any previous block in subsequent blocks.
![image](https://github.com/Budibase/budibase/assets/39269380/0a1dc023-7d8d-4e1e-9969-5ef5e87dfe26)

**Comment by Camherb** [comment]
Hi @mikesealey 

Thank you so much for the quick response! I have a much better understanding of the process now and how to access the form fields. I have another question about the form itself. I am a bit confused about the type of form I should be creating. I created a form for viewing data but this only allowed me to have fields that exist as columns in my Google sheet (Company Name, Date Created, Latitude, Longitude, etc.). I turned off all of the fields I didn't need except for one which I renamed Radius, but I am now realizing this is not the best way to go about it. Should I instead be making a create or update form? To give a better understanding of my intentions with this application, I am looking to create a form, that takes the date created, latitude, longitude, and radius, then displays the names of all the companies within the radius of the latitude and longitude for that date.
